internal/services/sources_service.go

```diff
@@ -42,7 +49,15 @@ func GetSource(w http.ResponseWriter, r *http.Request) {
 
 	resp, err := client.ShowSourceWithResponse(r.Context(), sources.ID(id), AddIdentityHeader)
 	if err != nil {
-		renderError(w, r, payloads.New3rdPartyClientError(r.Context(), "show source", err))
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "show source", err, resp.StatusCode()))
+		return
+	}
+	if resp.StatusCode() == 404 {
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "source not found", err, resp.StatusCode()))
+		return
+	}
+	if resp.StatusCode() < 200 || resp.StatusCode() > 299 {
+		renderError(w, r, payloads.SourcesClientError(r.Context(), "sources client error", err, resp.StatusCode()))
```

Comment:
I don't like it, I think it makes the code less readable, do you have a different suggestion?

Reply:
You can ether check the first character of Status or compare the StatusCode just like that. I would probably prefer >= 300 which is a bit cleaner than 299. Just go ahead and create a helper function in services/status_check.go something like `IsStatus2xx(code int)` and use that :)

Just do not introduce any `utils` or `helpers` package please, we already had one sneaked in in routes/helpers and I hate it already :)

Reply:
I see the same pattern you use both in services and client packages, therefore I suggest to create a brand new package "parsing". I think we could also move the function ParseInt64 there as it is similar problem kind of.